plan.go

```diff
@@ -2837,17 +2839,20 @@ func (r *emptyFieldsPlan) do(ctx *execCtx, f func(id interface{}, data []interfa
 
 type wrapFilterPlan struct {
 	*filterDefaultPlan
+	once  sync.Once
```

Comment:
That clarifies things . Since you said there will be a separate instance for `execCtx` for each goroutine. 

We don't even need `sync.Once` instead keeping a `map[*selectStmt]bool` is enough.

One question though, I dont see the ned to guard the map wih `sync.RWMutex` from your explanation it doesnt seem like there will be any concurrent use of the same `execCtx` instance.